runtime/lua/vim/diagnostic.lua

```diff
@@ -895,22 +901,31 @@ M.handlers.signs = {
     end
 
     local ns = M.get_namespace(namespace)
-    if not ns.user_data.sign_group then
-      ns.user_data.sign_group = string.format('vim.diagnostic.%s', ns.name)
+    if not ns.user_data.sign_ns then
+      local sign_ns = string.format('vim.diagnostic.%s', ns.name)
+      ns.user_data.sign_ns = vim.api.nvim_create_namespace(sign_ns)
```

Comment:
I know @clason suggested using a name for `sign_ns`, which I have no objection to. But to remain consistent we should do the same for the `virt_text_ns` and `underline_ns` namespaces (which are currently anonymous).